src/test/regress/expected/rpt.out

```diff
@@ -1613,6 +1663,210 @@ drop table t1;
+ Optimizer: Postgres-based planner
+ Settings: enable_bitmapscan = 'off', enable_seqscan = 'off'
+(16 rows)
+
+select count(*) from (select i from t group by i having f(i) > 0) i;
+ count 
+-------
+    10
+(1 row)
+
+-- ensure we do not make broadcast motion
+explain (costs off, verbose) select * from t1 where a in (select random() from t where i=a group by i);
+                              QUERY PLAN                               
+-----------------------------------------------------------------------
+ Gather Motion 3:1  (slice1; segments: 3)
+   Output: t1.a
+   ->  Seq Scan on rpt.t1
+         Output: t1.a
+         Filter: (SubPlan 1)
+         SubPlan 1
+           ->  Result
+                 Output: random(), t.i
+                 Filter: (t.i = t1.a)
+                 ->  Materialize
+                       Output: t.i
+                       ->  Broadcast Motion 1:3  (slice2; segments: 1)
+                             Output: t.i
+                             ->  Seq Scan on rpt.t
+                                   Output: t.i
+ Optimizer: Postgres-based planner
+ Settings: enable_bitmapscan = 'off', enable_seqscan = 'off'
+(17 rows)
+
+explain (costs off, verbose) select * from t1 where a in (select random() from t where i=a);
+                              QUERY PLAN                               
+-----------------------------------------------------------------------
+ Gather Motion 3:1  (slice1; segments: 3)
+   Output: t1.a
+   ->  Seq Scan on rpt.t1
+         Output: t1.a
+         Filter: (SubPlan 1)
+         SubPlan 1
+           ->  Result
+                 Output: random()
+                 Filter: (t.i = t1.a)
+                 ->  Materialize
+                       Output: t.i
+                       ->  Broadcast Motion 1:3  (slice2; segments: 1)
+                             Output: t.i
+                             ->  Seq Scan on rpt.t
+                                   Output: t.i
+ Optimizer: Postgres-based planner
+ Settings: enable_bitmapscan = 'off', enable_seqscan = 'off'
+(17 rows)
+
+-- ensure we make broadcast motion when volatile function in deleting motion flow
+explain (costs off, verbose) insert into t2 (a, b) select i, random() from t;
+                         QUERY PLAN                          
+-------------------------------------------------------------
+ Insert on rpt.t2
+   ->  Broadcast Motion 1:3  (slice1; segments: 1)
+         Output: t.i, (random())
+         ->  Result
+               Output: t.i, (random())
+               ->  Seq Scan on rpt.t
+                     Output: t.i, random()
+ Optimizer: Postgres-based planner
+ Settings: enable_bitmapscan = 'off', enable_seqscan = 'off'
+(9 rows)
+
+-- ensure we make broadcast motion when volatile function in correlated subplan qual
+explain (costs off, verbose) select * from t1 where a in (select f(i) from t where i=a and f(i) > 0);
+                                QUERY PLAN                                
```

Comment:
This plan doesn't check correlated predicate and as a result may produce invalid output.